x/interchainstaking/keeper/ibc_packet_handlers_test.go

```diff
@@ -2785,3 +2787,75 @@ func (suite *KeeperTestSuite) TestTriggerRedemptionRate() {
 		suite.Equal(prevAllBalancesQueryCnt+1, allBalancesQueryCnt)
 	})
 }
+func (suite *KeeperTestSuite) TestHandleCompleteSend() {
+	suite.Run("TestHandleCompleteSend", func() {
+		suite.SetupTest()
+		suite.setupTestZones()
+
+		quicksilver := suite.GetQuicksilverApp(suite.chainA)
+		ctx := suite.chainA.GetContext()
+		ctx = ctx.WithContext(context.WithValue(ctx.Context(), utils.ContextKey("connectionID"), suite.path.EndpointA.ConnectionID))
+		zone, found := quicksilver.InterchainstakingKeeper.GetZone(ctx, suite.chainB.ChainID)
+		if !found {
+			suite.Fail("unable to retrieve zone for test")
+		}
+		quicksilver.InterchainstakingKeeper.IBCKeeper.ChannelKeeper.SetChannel(ctx, "transfer", "channel-0", TestChannel)
+		complete := time.Now().UTC()
+
+		// trigger handler with testcases
+		testCases := []struct {
+			name          string
+			msg           banktypes.MsgSend
+			memo          string
+			expectedError error
+		}{
+			{
+				name: "unexpected completed send",
+				msg: banktypes.MsgSend{
+					FromAddress: "",
+					ToAddress:   "",
+					Amount:      sdk.NewCoins(sdk.NewCoin(zone.BaseDenom, sdk.NewInt(1_000_000))),
+				},
+				expectedError: errors.New("unexpected completed send (2) from  to  (amount: 1000000uatom)"),
+			},
+			{
+				name: "is withdrawal address",
+				msg: banktypes.MsgSend{
+					FromAddress: zone.WithdrawalAddress.Address,
+					ToAddress:   "",
+					Amount:      sdk.NewCoins(sdk.NewCoin(zone.BaseDenom, sdk.NewInt(1_000_000))),
+				},
+				expectedError: nil,
+			},
+			{
+				name: "is to delegation address",
+				msg: banktypes.MsgSend{
+					FromAddress: zone.DepositAddress.Address,
+					ToAddress:   zone.DelegationAddress.Address,
+					Amount:      sdk.NewCoins(sdk.NewCoin(zone.BaseDenom, sdk.NewInt(1_000_000))),
+				},
+				memo:          fmt.Sprintf("unbondSend/%d", complete.Unix()),
```

Comment:
this is an invlaid memo for a send from deposit -> delegation (it will probably work, but the unbondSend/xxxx memo is always from delegate -> recipient of unbonded tokens)